docs - add best practices page for resource groups

lisakowen opened the pull request:
create a best practices page for resource groups
- base level of content, tried to provide info parallel to what is discussed on resource queue best practices page
- add a section describing admin utilities and admin_group resource group concurrency

doc review site link:
- http://docs-gpdb-review-staging.cfapps.io/review/best_practices/resgroups.html

gaos1 commented on gpdb-doc/dita/best_practices/resgroups.xml:
```diff
@@ -0,0 +1,177 @@
+            group named <codeph>admin_group</codeph>. Keep this in mind when
+            scheduling and running Greenplum administration utilities.</li>
+        <li>Ensure that each role is assigned a resource group. If you do not assign
+            a resource group to a role, queries submitted by the role are handled by the
```
to be more accurate, "a role" here should be  "a non-admin role".

gaos1 commented:
LGTM. @pivotal-ning-yu  could you also help to review the updates?

lisakowen commented:
thank you for reviewing, @gaos1 !

dyozie commented on gpdb-doc/dita/best_practices/resgroups.xml:
```diff
@@ -0,0 +1,177 @@
+<!DOCTYPE topic PUBLIC "-//OASIS//DTD DITA Topic//EN" "topic.dtd">
+<topic id="topic_hhc_z5w_r4">
+  <title>Memory and Resource Management with Resource Groups</title>
+  <shortdesc>Managing Greenplum Database resources with resource groups</shortdesc>
```
Add period to end of the shortdesc.

dyozie commented on gpdb-doc/dita/best_practices/resgroups.xml:
```diff
@@ -0,0 +1,177 @@
+        many out-of-memory conditions by configuring resource groups to manage expected
+        workloads. </p>
+      <p>The following operating system and Greenplum Database memory settings are
+        significant when you manage your Greenplum Database resources with resource groups:</p>
```
delete "your" here

dyozie commented on gpdb-doc/dita/best_practices/resgroups.xml:
```diff
@@ -0,0 +1,177 @@
+      <ul id="ul_q4x_g1g_zt">
+        <li>Total RAM = 256GB </li>
+        <li>Swap = 64GB </li>
+        <li>8 primary segments and 8 mirror segments per host, in blocks of 4 hosts.</li>
```
Remove period in this and next list item.

dyozie commented on gpdb-doc/dita/best_practices/resgroups.xml:
```diff
@@ -0,0 +1,177 @@
+    <section id="configuring_rg">
+      <title>Configuring Resource Groups</title>
+      <p>Greenplum Database resource groups provide a powerful mechanism for managing the workload
+        of the cluster. You can use resource groups to limit both the number of concurrent
```
I'd keep the intro here, but the follow-on sentences just repeat the main overview.  Maybe just replace with something like "Consider these general guidelines when you configure RGs for your system:" or similar.

dyozie commented on gpdb-doc/dita/best_practices/resgroups.xml:
```diff
@@ -0,0 +1,177 @@
+          group for the workload and the time of day.</li>
+        <li>Use the <codeph>gptoolkit</codeph> views to examine resource group resource
+            usage and to monitor how the groups are working.</li>
+      </ul>
```
This seems like a good spot to advertise Command Center capabilities around RGs (conditionalized, of course).

dyozie commented on gpdb-doc/dita/best_practices/resgroups.xml:
```diff
@@ -0,0 +1,177 @@
+        The default <codeph>CONCURRENCY</codeph> value for the <codeph>admin_group</codeph>
+        resource group is 10.</p>
+      <p>Certain Greenplum Database administrative utilities may use more than one
+        <codeph>CONCURRENCY</codeph> slot at runtime; for example, <codeph>gpbackup</codeph>
```
runtime; for example, -> runtime, such as

lisakowen commented:
@gaos1 , did you want someone from your team to review this, or are we good to merge?

pivotal-ning-yu commented:
Sorry, almost missed this PR. I'll review it now.

pivotal-ning-yu commented on gpdb-doc/dita/best_practices/resgroups.xml:
```diff
@@ -0,0 +1,180 @@
+        <li>When you specify a <codeph>memory_spill_ratio</codeph> in your session,
+            individual queries running in the session can allocate more than their "share"
+           of memory, which reduces the memory available for other queries running in the
+           resource group.</li>
```
Not sure if I understand above paragraph correctly. The session level `memory_spill_ratio` actually behave the same as the resource group option `MEMORY_SPILL_RATIO`, it doesn't change their share or slot quota, and doesn't reduce the memory available for other queries.  Increasing it could let current session allocate more memory (if not enough in slot quote then allocate from group shared memory and global shared memory); decreasing it could let current session allocate less memory, however even if there is unused memory in the slot quota they can not be used by other queries.

lisakowen commented:
@pivotal-ning-yu - thank you for reviewing.  i removed the memory_spill_ratio bullet.  i also clarified the calculations; please check it out and let me know if i addressed your concerns.

http://docs-gpdb-review-staging.cfapps.io/review/best_practices/resgroups.html#topic_hhc_z5w_r4__section_upm_rbl_zt

pivotal-ning-yu commented:
Looks perfect to me now, thanks very much.

On Sat, Aug 25, 2018 at 2:42 AM, Lisa Owen <notifications@github.com> wrote:

> @pivotal-ning-yu <https://github.com/pivotal-ning-yu> - thank you for
> reviewing. i removed the memory_spill_ratio bullet. i also clarified the
> calculations; please check it out and let me know if i addressed your
> concerns.
>
> http://docs-gpdb-review-staging.cfapps.io/review/best_
> practices/resgroups.html#topic_hhc_z5w_r4__section_upm_rbl_zt
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/greenplum-db/gpdb/pull/5430#issuecomment-415847422>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AYjBHgXcee79W1SNvGTxyq_4cddEuwsyks5uUEkKgaJpZM4V0dBS>
> .
>
